mrmustard/lab_dev/circuit_components.py

```diff
@@ -373,35 +403,65 @@ def on(self, modes: Sequence[int]) -> CircuitComponent:
+        available, otherwise it defaults to the value of ``AUTOCUTOFF_MAX_CUTOFF`` in the settings.
 
-        Uses the :meth:`mrmustard.physics.converters.to_fock` method to convert the internal
-        representation.
+        Args:
+            shape: The shape of the returned representation. If ``shape`` is given as an ``int``,
+                it is broadcasted to all the dimensions. If not given, it is estimated.
+        Returns:
+            array: The Fock representation of this component.
+        """
+        if isinstance(shape, int):
+            shape = (shape,) * self.representation.ansatz.num_vars
+        shape = shape or self.auto_shape()
+        assert len(shape) == len(self.fock_shape)
+        try:
+            As, bs, cs = self.bargmann
+            array = [math.hermite_renormalized(A, b, c, shape) for A, b, c in zip(As, bs, cs)]
+        except AttributeError:
+            array = self.representation.reduce(shape).array
+        return array
+
+    def to_fock(self, shape=None):
+        r"""
+        Returns a new circuit component with the same attributes as this and a ``Fock`` representation.
 
         .. code-block::
 
-            >>> from mrmustard.physics.converters import to_fock
             >>> from mrmustard.lab_dev import Dgate
+            >>> from mrmustard.physics.representations import Fock
 
             >>> d = Dgate([1], x=0.1, y=0.1)
             >>> d_fock = d.to_fock(shape=3)
 
             >>> assert d_fock.name == d.name
             >>> assert d_fock.wires == d.wires
-            >>> assert d_fock.representation == to_fock(d.representation, shape=3)
+            >>> assert isinstance(d_fock.representation, Fock)
 
         Args:
             shape: The shape of the returned representation. If ``shape``is given as
                 an ``int``, it is broadcasted to all the dimensions. If ``None``, it
-                defaults to the value of ``AUTOCUTOFF_MAX_CUTOFF`` in the settings.
+                defaults to the value of ``AUTOSHAPE_MAX`` in the settings.
         """
-        return self.__class__._from_attributes(
-            to_fock(self.representation, shape=shape),
-            self.wires,
-            self.name,
-        )
+        fock = Fock(math.astensor(self.fock(shape)), batched=True)
+        fock._original_bargmann_data = self.representation.data
+        return self._from_attributes(fock, self.wires, self.name)
+
+    def auto_shape(self) -> tuple[int, ...]:
+        r"""
+        The shape of the Fock representation of this component. If the component has a Fock representation
+        then it is just the shape of the array. If the components is a State in Bargmann
+        representation the shape is calculated using autoshape using the single-mode marginals.
+        If the component is not a State then the shape is a tuple of ``AUTOSHAPE_MAX``.
+        """
+        MAX = settings.AUTOSHAPE_MAX
```

Comment:
Well, the thing is that we need a full shape in order to create the array for the Fock representation. So all the `None`s in `fock_shape` should be replaced by some integer. If we do not know any better (because of the properties of the component, or because of its neighbours if we are in a circuit) then we have to pick _some_ default value, and this is the one. It's not good nor bad 😅

Reply:
I don't think we have to pick a default value. If we don't have a good/informed recommendation, we should just ask the user to kindly provide what _they_ think is a good shape, as I suspect they'll know better than this, no?

Reply:
Yes they may know better (not always). And in fact they are free to change the default by modifying `.fock_shape`, it's just that the completely hands-off experience still works if there is a last-resort default.